### PR TITLE
[Compute] `az image builder`: Add parameter `--staging-resource-group` in `create` command to support custom resource group naming and add subgroup `validator` to manage validate information of template

### LINKED ISSUE #22751: Azure Image Builder API 2022-02-14 Update
**Resource Provider** 
Azure VM Image Builder

**Description of Feature or Work Requested**
We now support VM Validation and custom resource group naming.

changes:

1. 1 new parameter for `az image builder create`
* `--stagingResourceGroup <string>`, type string, example inputs: "Custom_RG_Name001"

2. 3 new commands
*`az image builder validator`, get's validate information from template
*`az image builder validator add`, adds validate object to template
*`az image builder validator remove`, removes validate object from template
this includes 2 parameters
*`--continueDistributeOnFailure <boolean>`, type boolean (default false)
*`--sourceValidationOnly <boolean>`, type boolean (default false)

**Minimum API Version Required**
2022-02-14

**Swagger Link**
https://github.com/Azure/azure-rest-api-specs/pull/18896

**Target Date**
ASAP

### PR DESCRIPTION
**Pull request by yanzhudd**
**Description**

Add parameter `--staging-resource-group` for `az image builder create` to support custom resource group naming.
Add subgroup `az image builder validator` to manage validate information of template.
Close #22751

**Testing Guide**

**History Notes**

[Compute] `az image builder create`: Add parameter `--staging-resource-group` to support custom resource group naming
[Compute] `az image builder validator`: Add subgroup to manage validate information of template

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_help.py [hunk 1]
```diff
@@ -461,6 +461,51 @@
...
 
+helps['image builder validator'] = """
+type: group
+short-summary: Manage image builder template validate.
+"""
+
+helps['image builder validator add'] = """
+type: command
+short-summary: Add validate to an existing image builder template.
+long-summary: Must be used with --defer
```
[the remark this answers is not in the document]
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/_help.py:472]: > May I ask why do these commands have to be used with `--defer`?

It seems that the update operation (update an existing image template) is not supported in service side, so the parameter `--defer` is used to support managing validate in local cache, and then customer can choose to update the local cache to service to get the actual resource.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -350,6 +351,9 @@ def load_arguments(self, _):
         c.argument('file_source', arg_type=ib_file_customizer_type, help="The URI of the file to be downloaded into the image. It can be a github link, SAS URI for Azure Storage, etc.")
         c.argument('dest_path', arg_type=ib_file_customizer_type, help="The absolute destination path where the file specified in --file-source will be downloaded to in the image")
 
+    with self.argument_context('image builder validator add', min_api='2022-02-14') as c:
+        c.argument('dis_on_failure', options_list=['--continue-distribute-on-failure', '--dis-on-failure'], arg_type=get_three_state_flag(), help="If validation fails and this field is set to false, output image(s) will not be distributed.")
+        c.argument('source_validation_only', arg_type=get_three_state_flag(), help="If this field is set to true, the image specified in the 'source' section will directly be validated. No separate build will be run to generate and then validate a customized image.")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:356]:
```suggestion
        c.argument('dis_on_failure', options_list=['--continue-distribute-on-failure', '--dis-on-failure'], arg_type=get_three_state_flag(), help="If validation fails and this parameter is set to false, output image(s) will not be distributed.")
        c.argument('source_validation_only', arg_type=get_three_state_flag(), help="If this parameter is set to true, the image specified in the 'source' section will directly be validated. No separate build will be run to generate and then validate a customized image.")
```

File: src/azure-cli/azure/cli/command_modules/vm/_help.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/vm/_help.py:472]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**